src/main/osd/osd_warnings.c

```diff
@@ -277,32 +330,29 @@ void renderOsdWarning(char *warningText, bool *blinking, uint8_t *displayAttr)
 #ifdef USE_ESC_SENSOR
     // Show warning if we lose motor output, the ESC is overheating or excessive current draw
     if (featureIsEnabled(FEATURE_ESC_SENSOR) && osdWarnGetState(OSD_WARNING_ESC_FAIL) && ARMING_FLAG(ARMED)) {
-        char* p = warningText;
-        strcpy(p, "ESC");
-        p += strlen("ESC");
+        uint8_t escErrorLength = 0;
+        
+        // Write 'ESC' prefix
+        escErrorLength += tfp_sprintf(warningText + escErrorLength, "ESC");
 
         bool escWarning = false;
-        for (unsigned i = 0; i < getMotorCount() && p < warningText + OSD_WARNINGS_MAX_SIZE - 1; i++) {
+        for (unsigned i = 0; i < getMotorCount() && escErrorLength < OSD_WARNINGS_MAX_SIZE - 6; i++) {
```

Comment:
You can handle the overflow when appending motor. And the buffer has one extra byte for `'\0'`
```suggestion
        for (unsigned i = 0; i < getMotorCount() && escErrorLength < OSD_WARNINGS_MAX_SIZE; i++) `{`
```